File: Web/RMA-Web-Shell/projects/claimcare/src/app/claim-manager/shared/entities/funeral/ClaimPool.ts
import { STPExitReasonEnum } from "projects/shared-models-lib/src/lib/enums/stp-exit-reason.enum";
import { InjurySeverityTypeEnum } from "../../enums/injury-severity-type-enum";
import { ClaimLiabilityStatusEnum } from "projects/shared-models-lib/src/lib/enums/claim-liability-status.enum";
import { ClaimStatusEnum } from "projects/shared-models-lib/src/lib/enums/claim-status.enum";
import { EventTypeEnum } from "../../enums/event-type-enum";


export class ClaimPool {
  underwriterId: number;
  pdPercentage: number;
  instruction: string;
  claimId: number;
  eventId: number;
  personEventId: number;
  claimNumber: string;
  personEventReference: string;
  eventType: EventTypeEnum;
  assignedTo?: number;
  claimStatus: ClaimStatusEnum;
  liabilityStatus: ClaimLiabilityStatusEnum;
  diseaseDescription: string;
  dateCreated?: Date;
  priority: string;
  memberName: string;
  insuredLife: string;
  identificationNumber: string;
  personEventCreatedBy: string;
  lastModifiedBy: string;
  injuryType: InjurySeverityTypeEnum;
  employeeNumber: string;
  employeeIndustryNumber: string;
  lastWorkedOnUserId?: number;
  isTopEmployer: boolean;
  included: boolean;
  companyName: string;
  companyReferenceNumber: string;
  stpExitReason?: STPExitReasonEnum;
  stpExitReasonDescription?: string;
  userName: string;
  userId?: number
  application: string;
  bucketClassId: number;
  investigationRequired: boolean;
  employerRolePlayerId: number;
  employeeRolePlayerId: number;
}

File: Service/ClaimCare/RMA.Service.ClaimCare.Contracts/Entities/ClaimPool.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;
using RMA.Service.ClaimCare.Contracts.Enums;

using System;

namespace RMA.Service.ClaimCare.Contracts.Entities
{
    public class ClaimPool
    {
        public int? UnderwriterId { get; set; }
        public int? PdPercentage { get; set; }
        public string Instruction { get; set; }
        public int? ClaimId { get; set; }
        public int EventId { get; set; }
        public int PersonEventId { get; set; }
        public string ClaimNumber { get; set; }
        public string PersonEventReference { get; set; }
        public EventTypeEnum EventType { get; set; }
        public int? AssignedTo { get; set; }
        public ClaimStatusEnum? ClaimStatus { get; set; }
        public ClaimLiabilityStatusEnum? LiabilityStatus { get; set; }
        public string DiseaseDescription { get; set; }
        public DateTime? DateCreated { get; set; }
        public string Priority { get; set; }
        public string InsuredLife { get; set; }
        public string IdentificationNumber { get; set; }
        public string PersonEventCreatedBy { get; set; }
        public string LastModifiedBy { get; set; }
        public InjurySeverityTypeEnum InjuryType { get; set; }
        public string EmployeeNumber { get; set; }
        public string EmployeeIndustryNumber { get; set; }
        public int? LastWorkedOnUserId { get; set; }
        public bool IsTopEmployer { get; set; }
        public bool Included { get; set; }
        public string CompanyName { get; set; }
        public string CompanyReferenceNumber { get; set; }
        public int BucketClassId { get; set; }
        public int? STPExitReason { get; set; }
        public string STPExitReasonDescription { get; set; }
        public string UserName { get; set; }
        public int? UserId { get; set; }
        public string Application { get; set; }
        public bool? InvestigationRequired { get; set; }
        public int IsFatal { get; set; }
        public int EmployerRolePlayerId { get; set; }
        public int EmployeeRolePlayerId { get; set; }
}
}